mixer/pkg/api/grpcServer_test.go

```diff
@@ -254,6 +261,82 @@ func TestCheck(t *testing.T) {
 	}
 }
 
+func TestCheckCachedDenial(t *testing.T) {
+	ts, err := prepTestState()
+	if err != nil {
+		t.Fatalf("Unable to prep test state: %v", err)
+	}
+	defer ts.cleanupTestState()
+
+	ts.check = func(ctx context.Context, requestBag attribute.Bag) (*adapter.CheckResult, error) {
+		return &adapter.CheckResult{
+			Status:        status.WithPermissionDenied("Not Implemented"),
+			ValidDuration: time.Hour * 1000,
+		}, nil
+	}
+
+	attr0 := mixerpb.CompressedAttributes{
+		Words: []string{"A1", "A2", "A3"},
+		Int64S: map[int32]int64{
+			-1: 25,
+			-2: 26,
+			-3: 27,
+		},
+	}
+
+	request := mixerpb.CheckRequest{Attributes: attr0}
+
+	// do this twice to try out cached denials
+	for i := 0; i < 2; i++ {
+		cr, err := ts.client.Check(context.Background(), &request)
+		if err != nil {
+			t.Errorf("Expecting success, got %v", err)
+		}
+
+		if status.IsOK(cr.Precondition.Status) {
+			t.Errorf("Expecting error, got OK")
+		}
+	}
+}
+
+func TestCheckCachedAllow(t *testing.T) {
+	ts, err := prepTestState()
+	if err != nil {
+		t.Fatalf("Unable to prep test state: %v", err)
+	}
+	defer ts.cleanupTestState()
+
+	ts.check = func(ctx context.Context, requestBag attribute.Bag) (*adapter.CheckResult, error) {
+		return &adapter.CheckResult{
+			Status:        status.OK,
+			ValidDuration: time.Hour * 1000,
+		}, nil
+	}
+
+	attr0 := mixerpb.CompressedAttributes{
```

Comment:
this is used in both new tests. Should this be extracted out and augmented with comments (what are 25, 26, and 27?)?